spec/sec-string-objects-patch.html

```diff
@@ -1,39 +1,74 @@
 <emu-clause id="sec-properties-of-the-string-prototype-object">
   <h1>Properties of the String Prototype Object</h1>
   <emu-clause id="sec-string.prototype.match">
-    <h1>String.prototype.match ( _regexp_ <ins>[, _mapfn_ ]</ins> )</h1>
-    <p>When the `match` method is called with argument _regexp_ <ins>and optional argument _mapfn_</ins>, the following steps are taken:</p>
+    <h1>String.prototype.match ( _regexp_ <ins>[, _options_ ]</ins> )</h1>
+    <p>When the `match` method is called with argument _regexp_ <ins>and optional argument _options_</ins>, the following steps are taken:</p>
     <emu-alg>
     1. Let _O_ be ? RequireObjectCoercible(*this* value).
-    1. <ins>If _mapfn_ is not *undefined* and IsCallable(_mapfn_) is *false*, throw a *TypeError* exception.</ins>
+    1. <ins>If _options_ is not present, let _options be *undefined*.</ins>
+    1. <ins>Let _matchOptions_ be ? ToMatchOptions(_options_).</ins>
```

Comment:
any reason not to do FromMatchOptions here?
```suggestion
    1. <ins>Let _matchOptions_ be ! FromMatchOptions(? ToMatchOptions(_options_)).</ins>
```

Reply:
Because Steps 3.1, 4, and 5 could ReturnIfAbrupt before the MatchOptions needs to be turned back into an object, so there is no sense in adding additional overhead for object allocation if the result is never used.

Reply:
That seems like an unobservable optimization implementations could apply, and not something that needs to be baked into the algorithm?